src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -7146,11 +7154,11 @@ internal static List<CompletionResult> CompleteHashtableKey(CompletionContext co
                         {
                             case "New-Object":
                                 var inferredType = AstTypeInference.InferTypeOf(commandAst, completionContext.TypeInferenceContext, TypeInferenceRuntimePermissions.AllowSafeEval);
-                                var result = new List<CompletionResult>();
```

Comment:
Why change this? None of the variables exit the declaration scope as far as I can see.

Reply:
Because it was a little silly to have both a "result" and "results" variable when you can have just one. What's the drawback?

Reply:
Locality rule - variables should be declared at innermost scope, preferably at initialization. And you only ever used one of them, they never interacted. If I see a variable predeclared, I expect that the current scope will use assignment from an inner one:

```csharp
int i = 0;
if (condition) {
    i = ...;
    // use i
} else {
    i = bar();
    // use i
}
// Because author predeclared `i`, I will now look for its usage here
```

If you only wanted to unify the names, which is valid to me, you can do it without predeclaring.

Reply:
I cannot. results is used both inside and outside the if statements so the best location I can declare it is outside the if statements. It would probably work if I wrap it inside a silly if statement like `if (true)` but that's IMO a worse solution than just declaring it before all the if statements.

Reply:
Okay, I may have missed the outside usage